src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -88,6 +88,29 @@ bool is_any_user_cpu(const std::list<const program_node*>& users) {
     }
     return false;
 }
+
+kernel_impl_params primitive_impl::static_canonicalize_shapes(const kernel_impl_params& impl_params) {
+        auto updated_impl_params = impl_params;
```

Comment:
Fused ops handling needs to be added here especially for eltwise. Currently we allow eltwise fusion only when eltwise_out_rank == prim_out_rank, but additional input needs to be broadcasted properly. I think something as follows is required:
```cpp
        for (auto& fd : updated_impl_params.fused_desc) {
            if (fd.is_type<eltwise>() && fd.total_num_deps == 2) {
                auto out_rank = updated_impl_params.output_layouts[0].get_partial_shape().size();

                auto dep_idx = fd.dep_start_idx;
                auto dep_shape = updated_impl_params.input_layouts[dep_idx].get_partial_shape();
                // numpy broadcasting
                dep_shape.insert(dep_shape.begin(), out_rank - dep_shape.size(), ov::Dimension(1));
                updated_impl_params.input_layouts[dep_idx].set_partial_shape(dep_shape);
            }
        }
```